pkg/buses/natss/bus.go

```diff
@@ -0,0 +1,178 @@
+			return nil
+		},
+		SubscribeFunc:   bus.subscribe,
+		UnsubscribeFunc: bus.unsubscribe,
+	}
+	setup(bus)
+	return bus, nil
+}
+
+func (b *NatssBus) Run(threadness int, stopCh <-chan struct{}, clientId string) {
+	b.logger.Infof("try to connect to NATSS from %q", clientId)
+	var err error
+	for i := 0; i < 60; i++ {
+		if natsConn, err = stanutil.Connect("knative-nats-streaming", clientId, b.natssUrl, b.logger); err != nil {
+			b.logger.Errorf(" Create new connection failed: %+v", err)
+			time.Sleep(1 * time.Second)
+		} else {
+			break
+		}
+	}
+	if err != nil {
+		b.logger.Errorf(" Create new connection failed: %+v", err)
+		return
+	}
+	b.logger.Info("connection to NATSS established, natsConn=%+v", natsConn)
+
+	if b.dispatcher != nil {
+		b.dispatcher.Run(threadness, stopCh)
+	}
+	if b.provisioner != nil {
+		b.provisioner.Run(threadness, stopCh)
+	}
+}
+
+func SetNewBusProvisioner(opts *buses.BusOpts) SetupNatssBus {
+	return func(b *NatssBus) error {
+		b.natssUrl = NatssUrl
+		b.provisioner = buses.NewBusProvisioner(b.ref, b.provEventHandler, opts)
+		b.logger = opts.Logger
+		return nil
+	}
+}
+
+func SetNewBusDispatcher(opts *buses.BusOpts) SetupNatssBus {
+	return func(b *NatssBus) error {
+		b.natssUrl = NatssUrl
+		b.dispatcher = buses.NewBusDispatcher(b.ref, b.dispEventHandler, opts)
+		b.logger = opts.Logger
+		return nil
+	}
+}
+
+func (bus *NatssBus) subscribe(channel buses.ChannelReference, subscription buses.SubscriptionReference, parameters buses.ResolvedParameters) error {
+	bus.logger.Infof("Subscribe %q to %q channel", subscription.Name, channel.Name)
+
+	mcb := func(msg *stan.Msg) {
+		bus.logger.Infof("NATSS message received: %+v", msg)
+		message := buses.Message{
+			Headers: map[string]string{},
+			Payload: []byte(msg.Data),
+		}
+		if err := bus.dispatcher.DispatchMessage(subscription, &message); err != nil {
+			bus.logger.Warnf("Failed to dispatch message: %v", err)
+			return
+		}
+		if err := msg.Ack(); err != nil {
+			bus.logger.Warnf("Failed to acknowledge message: %v", err)
+		}
+	}
+	// subscribe to a NATSS subject
+	if natsStreamingSub, err := (*natsConn).Subscribe(channel.Name, mcb, stan.DurableName(subscription.Name), stan.SetManualAckMode(), stan.AckWait(1*time.Minute)); err != nil {
+		bus.logger.Errorf(" Create new NATSS Subscription failed: %+v", err)
+		return err
+	} else {
+		bus.logger.Infof("NATSS Subscription created: %+v", natsStreamingSub)
+		bus.subscribers[subscription.Name] = &natsStreamingSub
+	}
+	return nil
+}
+
+func (bus *NatssBus) unsubscribe(channel buses.ChannelReference, subscription buses.SubscriptionReference) error {
+	bus.logger.Infof("Unsubscribe %q from %q channel", subscription.Name, channel.Name)
+
+	// unsubscribe from a NATSS subject
+	if natsStreamingSub, ok := bus.subscribers[subscription.Name]; ok {
+		if err := (*natsStreamingSub).Unsubscribe(); err != nil {
+			bus.logger.Errorf(" Unsubscribe() failed: %+v", err)
```

Comment:
`zap.Error(err)` for pretty print.

Also there is a space between `"` and `Unsubscribe()`

Reply:
also just `%v`, no `+`